src/parser/command.rs

```diff
@@ -656,4 +698,145 @@ mod tests {
             ]
         })
     );
+
+    make_test!(
+        defgate,
+        parse_defgate,
+        r#"H:
+            1/sqrt(2), 1/sqrt(2)
+            1/sqrt(2), -1/sqrt(2)"#,
+        Instruction::GateDefinition(GateDefinition {
+            name: "H".to_string(),
+            parameters: vec![],
+            matrix: vec![
+                vec![
+                    Expression::Infix {
```

Comment:
For brevity, you could define this once and clone it

Reply:
Don't believe that is possible. We can't initiailize a `Box` within a constant expression.

Reply:
Indeed it is! Check out #97 . The last macro argument is just an expression, which can be wrapped in a block and remain an expression - if that's what you mean in your comment.

Reply:
Oh! That's super cool.

My immediate thought was raising it to a `const` and reusing it. In that case, you can't use the `Box`. I never even considered a block that returns the final value. Very slick.